### PR TITLE
[EDU-1364] - Add & Update Existing Tests for the Rest/Realtime Selector & other related changes

### PR DESCRIPTION
**Pull request by aralovelace**
## Description

A PR description indicating the purpose of the PR.

* [EDU-1364](https://ably.atlassian.net/browse/EDU-1364)

**Update following tests:**
 - Pre Component :  Add rest and realtime languages in the attributes and add new tests with realtime and rest data
- LocalLanguageAlternative:  Add rest and realtime languages in the attributes and new tests make sure the SDKInterface menu is visible  if SDK languages exists
- LanguageButton: adding new tests to make sure SDK languages still supported and with localChange being removed is should still display a correct active button

**Add new test file**
- SDKInterfacePanel - add different scenarios to make sure this component is fully testes

## Review

Instructions on how to review the PR. 

*  Run the tests in your local

[EDU-1364]: https://ably.atlassian.net/browse/EDU-1364?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/blocks/software/Pre.test.tsx [hunk 1]
```diff
@@ -3,7 +3,35 @@ import { render, screen } from '@testing-library/react';
 import { PageLanguageContext } from 'src/contexts';
 import { Pre } from './';
 
-const PRE_TEST_DATA = [{ data: [{ data: 'useVars = false;', type: 'text' }], type: 'tag', name: 'code' }];
+const DATA_ARRAY_TEST = [{ data: 'useVars = false;', type: 'text' }];
+const PRE_TEST_SINGLE_DATA = { data: DATA_ARRAY_TEST, type: 'tag', name: 'code' };
```
Comment (artismarti) [inline src/components/blocks/software/Pre.test.tsx:7]: What does `single_data` mean here?

File: src/components/blocks/software/Pre.test.tsx [hunk 2]
```diff
@@ -42,4 +70,50 @@ describe('<Pre />', () => {
...
+        restAltData={{
+          rest_javascript: PRE_REST_JS_TEST_DATA,
+          rest_csharp: PRE_REST_CSHARP_TEST_DATA,
+        }}
+      />,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it('should successfully render code elements with  only Realtime languages', () => {
```
Comment (artismarti) [inline src/components/blocks/software/Pre.test.tsx:100]: Could we add a scenario here for Rest only as well, please?

**Review by artismarti — CHANGES_REQUESTED** [review]
Just a question & one minor request, please! :)

File: src/components/blocks/software/Pre.test.tsx (hunk 2 above)
Comment (aralovelace) [reply src/components/blocks/software/Pre.test.tsx:100]: @artismarti  pushed and added

**Review by artismarti — APPROVED** [review]

**Pull request merged**